### PR TITLE
PMM-4648 Rotate slowlog.

### PR DESCRIPTION
**Pull request by askomorokhov**
https://jira.percona.com/browse/PMM-4437

depends on:
https://github.com/percona/pmm-agent/pull/116
https://github.com/percona/pmm-managed/pull/264

FB:
https://github.com/Percona-Lab/pmm-submodules/pull/444

usage ex.:
`pmm-admin add mysql --username=root --password=secret ps:3306 MySQLSlowLog --size-slow-logs=10MB`

### CODE REVIEW

File: Gopkg.toml [hunk 1]
```diff
@@ -12,7 +12,8 @@ required = [
 # TODO switch to tags before PMM 2.0 release
 [[constraint]]
   name = "github.com/percona/pmm"
-  branch = "PMM-2.0"
+  #branch = "PMM-2.0"
+  branch = "PMM-4437-flush-slowlogs"
```
Comment (AlekSi) [inline Gopkg.toml]: Merged with changes.

File: Gopkg.toml [hunk 2]
```diff
@@ -12,7 +12,8 @@ required = [
...
```
Comment (BupycHuk) [inline Gopkg.toml]: it's merged, isn't it?

File: commands/inventory/add_agent_qan_mysql_slowlog_agent.go [hunk 3]
```diff
@@ -94,4 +98,5 @@ func init() {
 	AddAgentQANMySQLSlowlogAgentC.Flag("password", "MySQL password for scraping metrics").StringVar(&AddAgentQANMySQLSlowlogAgent.Password)
 	AddAgentQANMySQLSlowlogAgentC.Flag("custom-labels", "Custom user-assigned labels").StringVar(&AddAgentQANMySQLSlowlogAgent.CustomLabels)
 	AddAgentQANMySQLSlowlogAgentC.Flag("skip-connection-check", "Skip connection check").BoolVar(&AddAgentQANMySQLSlowlogAgent.SkipConnectionCheck)
+	AddAgentQANMySQLSlowlogAgentC.Flag("size-slow-logs", "Rotate slow logs. (0 = no rotation)").Default("1GB").BytesVar(&AddAgentQANMySQLSlowlogAgent.SizeSlowLogs)
```
Comment (AlekSi) [inline commands/inventory/add_agent_qan_mysql_slowlog_agent.go]: Inventory commands should not have default

File: commands/management/add_mysql.go [hunk 4]
```diff
@@ -114,6 +117,7 @@ func (cmd *addMySQLCommand) Run() (commands.Result, error) {
 			QANMysqlPerfschema: usePerfschema,
 
 			SkipConnectionCheck: cmd.SkipConnectionCheck,
+			SizeSlowLogs:        int64(cmd.SizeSlowLogs),
```
Comment (AlekSi) [inline commands/management/add_mysql.go]: We should use -1 or other negative value to disable rotation.

**Review by AlekSi — CHANGES_REQUESTED** [review]

**Review by BupycHuk — APPROVED** [review]

File: commands/inventory/add_agent_qan_mysql_slowlog_agent.go [hunk 5]
```diff
@@ -93,4 +115,6 @@ func init() {
 	AddAgentQANMySQLSlowlogAgentC.Flag("password", "MySQL password for scraping metrics").StringVar(&AddAgentQANMySQLSlowlogAgent.Password)
 	AddAgentQANMySQLSlowlogAgentC.Flag("custom-labels", "Custom user-assigned labels").StringVar(&AddAgentQANMySQLSlowlogAgent.CustomLabels)
 	AddAgentQANMySQLSlowlogAgentC.Flag("skip-connection-check", "Skip connection check").BoolVar(&AddAgentQANMySQLSlowlogAgent.SkipConnectionCheck)
+	AddAgentQANMySQLSlowlogAgentC.Flag("disable-queryexamples", "Disable collection of query examples").BoolVar(&AddAgentQANMySQLSlowlogAgent.DisableQueryExamples)
+	AddAgentQANMySQLSlowlogAgentC.Flag("size-slow-logs", "Rotate slow log file at this size (0 disables rotation)").BytesVar(&AddAgentQANMySQLSlowlogAgent.MaxSlowlogFileSize)
```
Comment (BupycHuk) [inline commands/inventory/add_agent_qan_mysql_slowlog_agent.go]: according what I've seen in pmm-managed 0 sets default value to 1GB
https://github.com/percona/pmm-managed/pull/264/files#diff-ce82fa739acbb612f2b2db458c067721R48

File: commands/management/add_mysql.go [hunk 6]
```diff
@@ -110,6 +113,11 @@ func (cmd *addMySQLCommand) Run() (commands.Result, error) {
 		}
 	}
 
+	// tweak according to API docs
```
Comment (BupycHuk) [inline commands/management/add_mysql.go]: it's confusing. Logic should be the same in API and in CLI.

File: commands/management/add_mysql.go [hunk 7]
```diff
@@ -188,6 +199,8 @@ func init() {
 	AddMySQLC.Flag("query-source", querySourceHelp).Default(querySources[0]).EnumVar(&AddMySQL.QuerySource, querySources...)
 	AddMySQLC.Flag("use-perfschema", "Run QAN perf schema agent").Hidden().BoolVar(&AddMySQL.UsePerfschema)
 	AddMySQLC.Flag("use-slowlog", "Run QAN slow log agent").Hidden().BoolVar(&AddMySQL.UseSlowLog)
+	AddMySQLC.Flag("disable-queryexamples", "Disable collection of query examples").BoolVar(&AddMySQL.DisableQueryExamples)
+	AddMySQLC.Flag("size-slow-logs", "Rotate slow log file at this size (0 disables rotation)").Default("1GB").BytesVar(&AddMySQL.MaxSlowlogFileSize)
```
Comment (BupycHuk) [inline commands/management/add_mysql.go]: let's add `default: 1GB` to help message

**Review by BupycHuk — CHANGES_REQUESTED** [review]

File: commands/inventory/add_agent_qan_mysql_slowlog_agent.go (hunk 5 above)
Comment (AlekSi) [reply commands/inventory/add_agent_qan_mysql_slowlog_agent.go]: Inventory API is dump – 0 or absence of value means 0.
Management API is smart – 0 or absence of value means "safe default" (1 GB).

This is `pmm-admin inventory` command. Linked pmm-managed code is Management API.

**Review by BupycHuk — APPROVED** [review]

**Pull request merged**